src/main/kotlin/ink/abb/pogo/scraper/tasks/LootOneNearbyPokestop.kt

```diff
@@ -89,14 +89,17 @@ class LootOneNearbyPokestop(val sortedPokestops: List<Pokestop>, val lootTimeout
         }
     }
 
-    // TODO: Does not work as everything is multithread and the rest of the bot just continues
-    private fun checkResult(result: PokestopLootResult) {
+    private fun checkResult(result: PokestopLootResult, bot: Bot) {
         if (result.experience == 0 && result.itemsAwarded.isEmpty()) {
-            Log.red("Looks like a ban. Pause for $pauseDuration minute(s).")
-            Thread.sleep(TimeUnit.MINUTES.toMillis(pauseDuration))
-            pauseDuration += 1
+            Log.red("Looks like a ban. Pause for $pauseDuration seconds.")
+            try {
+                // Wait <pauseDuration> seconds or until latch is at 0 -> bot is being stopped
+                bot.runningLatchAwait(pauseDuration, TimeUnit.SECONDS)
+            } catch (ignore: InterruptedException) {
```

Comment:
I would put this try-catch in the bot method - seems like it would have to be everywhere.